nmostesting/mocks/Registry.py

```diff
@@ -271,3 +427,204 @@ def heartbeat(version, node_id):
         return jsonify({"health": int(time.time())})
     else:
         abort(404)
+
+
+@REGISTRY_API.route('/x-nmos/query', methods=["GET"], strict_slashes=False)
+def query_root():
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    query_api_versions = ['v1.0/', 'v1.1/', 'v1.2/', 'v1.3/']
+    max_api_index = query_api_versions.index(str(registry.query_api_version + '/'))
+    base_data = query_api_versions[0:max_api_index + 1]
+
+    return Response(json.dumps(base_data), mimetype='application/json')
+
+
+@REGISTRY_API.route('/x-nmos/query/<version>', methods=["GET"], strict_slashes=False)
+def query(version):
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    base_data = ['devices/', 'flows/', 'nodes/', 'receivers/', 'senders/', 'sources/', 'subscriptions/']
+
+    return Response(json.dumps(base_data), mimetype='application/json')
+
+
+def compare_resources(resource1, resource2):
+    try:
+        return NMOSUtils.compare_resource_version(resource1['version'], resource2['version'])
+    except Exception as e:
+        print(e)
+        return 0
+
+
+@REGISTRY_API.route('/x-nmos/query/<version>/<resource>', methods=["GET"], strict_slashes=False)
+def query_resource(version, resource):
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    MIN_SINCE = "0:0"
+    MAX_UNTIL = NMOSUtils.get_TAI_time()
+
+    base_data = []
+
+    if request.args.get('paging.since') or request.args.get('paging.until'):
+        registry.pagination_used = True
+
+    since = request.args.get('paging.since') or MIN_SINCE
+    until = request.args.get('paging.until') or MAX_UNTIL
+    limit = min(int(request.args.get('paging.limit') or '100'), registry.paging_limit)
```

Comment:
Hey, @jonathan-r-thorpe, please would you explain what you're trying to achieve with the two different defaults?

Reply:
No excuse for that magic number (shame faced). It should be:

`limit = min(int(request.args.get('paging.limit') or registry.paging_limit), registry.paging_limit)`

allowing the mock registry to override the requested limit. But handling the case where no limit was explicitly requested

Reply:
Changed